app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/firestore/FirestoreCategorizedPicture.kt

```diff
@@ -13,9 +15,35 @@ data class FirestoreCategorizedPicture internal constructor(
     override val category: Category,
     val url: String,
 ) : CategorizedPicture, FirestoreDocument {
+    constructor(parcel: Parcel) : this(
+        parcel.readString()!!,
+        parcel.readParcelable(Category::class.java.classLoader)!!,
+        parcel.readString()!!
+    )
+
     override fun displayOn(activity: Activity, imageView: ImageView) {
         Glide.with(activity)
             .load(Firebase.storage.getReferenceFromUrl(url))
             .into(imageView)
     }
+
+    override fun writeToParcel(parcel: Parcel, flags: Int) {
+        parcel.writeString(path)
+        parcel.writeParcelable(category, flags)
+        parcel.writeString(url)
+    }
+
+    override fun describeContents(): Int {
+        return 0
+    }
+
+    companion object CREATOR : Parcelable.Creator<FirestoreCategorizedPicture> {
```

Comment:
Isn't the whole point of `kotlin-parcelize` to not have all of this boilerplate?

Reply:
I am not entirely sure. I will look into it.

Reply:
It does indeed, thanks!